eth2/beacon/chains/base.py

```diff
@@ -286,14 +286,27 @@ def get_state_machine_class_for_block_slot(
                 return sm_class
         raise StateMachineNotFound("No StateMachine available for block slot: #{0}".format(slot))
 
-    def get_state_machine(self, at_slot: Slot=None) -> 'BaseBeaconStateMachine':
+    def get_state_machine_by_block(self, block: BaseBeaconBlock=None) -> 'BaseBeaconStateMachine':
+        """
+        Return the ``StateMachine`` instance for the given block.
+        """
+        if block is None:
+            block = self.get_canonical_head()
+        sm_class = self.get_state_machine_class(block)
+
+        return sm_class(
+            chaindb=self.chaindb,
+            attestation_pool=self.attestation_pool,
+            slot=block.slot,
+            block=block,
+        )
+
+    def get_state_machine_by_slot(self, slot: Slot=None) -> 'BaseBeaconStateMachine':
         """
         Return the ``StateMachine`` instance for the given slot number.
```

Comment:
It's weird that the function is called `get_state_machine_by_slot` while `slot: Slot=None`. Maybe note in the docstring:
```suggestion
        Return the ``StateMachine`` instance for the given `slot`.
        If the given ``slot`` is ``None``, use the head slot number of the canonical chain.
```